feat(rollup-plugin-import-meta-assets): init plugin (#553)
modernweb-dev/web: docs/docs/building/rollup-plugin-import-meta-assets.md

---
title: Rollup Plugin import-meta-assets
eleventyNavigation:
  key: Rollup Plugin import-meta-assets
  parent: Building
  order: 2
---

> Rollup plugin that detects assets using `new URL('./path/to/asset.ext', import.meta.url)`, copies them and rewrite ES modules so relative paths still work after bundling.

Reviewer: ```suggestion
Rollup plugin that detects assets references relative to modules using patterns such as `new URL('./path/to/asset.ext', import.meta.url)`. The assets are added to the rollup pipeline, allowing them to be transformed and hash the filenames.
```

## Install

Using npm:

```
npm install @web/rollup-plugin-import-meta-assets --save-dev
```

## Usage

Create a rollup.config.js [configuration file](https://www.rollupjs.org/guide/en/#configuration-files) and import the plugin:

```js

> import { importMetaAssets } from '@web/rollup-plugin-import-assets';

Reviewer: ```suggestion
import { importMetaAssets } from '@web/rollup-plugin-import-meta-assets';
```

By default no files are ignored.

### `include`

Type: `String` | `Array[...String]`<br>
Default: `null`

A [picomatch pattern](https://github.com/micromatch/picomatch#globbing-features), or array of patterns, which specifies the files in the build the plugin should operate on.
By default all files are targeted.

### `warnOnError`

Type: `Boolean`<br>
Default: `false`

By default, the plugin quits the build process when it encounters an error. If you set this option to true, it will throw a warning instead and leave the code untouched.

### `transform`

Type: `Function`<br>
Default: `null`

By default, referenced assets detected by this plugin are just copied as is to the output directory, according to your configuration.

When `transform` is defined, this function will be called for each asset with two parameters, the content of the asset as a [Buffer](https://nodejs.org/api/buffer.html) and the absolute path.
This allows you to conditionnaly match on the absolute path and maybe transform the content.

In this example, we use it to optimize SVG images with [svgo](https://github.com/svg/svgo):

```js

> import { importMetaAssets } from '@web/rollup-plugin-import-assets';

Reviewer: ```suggestion
import { importMetaAssets } from '@web/rollup-plugin-import-meta-assets';
```
